Mage.Sets/src/mage/cards/c/CosmiumCatalyst.java

```diff
@@ -0,0 +1,98 @@
+package mage.cards.c;
+
+import java.util.UUID;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.filter.FilterCard;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.common.TargetCardInExile;
+import mage.util.CardUtil;
+
+/**
+ * Cosmium Catalyst
+ * Artifact
+ * {1}{R}, {T}: Choose an exiled card used to craft Cosmium Catalyst at random. You may cast that card without paying its mana cost.
+ *
+ * @author DominionSpy
+ */
+public class CosmiumCatalyst extends CardImpl {
+
+    public CosmiumCatalyst(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "");
+
+        this.nightCard = true;
+        this.color.setRed(true);
+
+        // {1}{R}, {T}: Choose an exiled card used to craft Cosmium Catalyst at random. You may cast that card without paying its mana cost.
+        Ability ability = new SimpleActivatedAbility(new CosmiumCatalystEffect(), new ManaCostsImpl<>("{1}{R}"));
+        ability.addCost(new TapSourceCost());
+        this.addAbility(ability);
+    }
+
+    protected CosmiumCatalyst(CosmiumCatalyst card) {
+        super(card);
+    }
+
+    @Override
+    public CosmiumCatalyst copy() {
+        return new CosmiumCatalyst(this);
+    }
+}
+
+class CosmiumCatalystEffect extends OneShotEffect {
+
+    public CosmiumCatalystEffect() {
```

Comment:
fyi the constructor can be package-private (no public access modifier) when the class is, as for all these card-specific custom classes, and the copy constructor private.